packages/kbn-monaco/src/esql/lib/esql_ast_provider.ts

```diff
@@ -41,7 +41,12 @@ export class ESQLAstAdapter {
   async validate(model: monaco.editor.ITextModel, code: string) {
     const getAstFn = await this.getAstWorker(model);
     const text = code ?? model.getValue();
-    const { errors, warnings } = await validateQuery(text, getAstFn, undefined, this.callbacks);
+    const { errors, warnings } = await validateQuery(
```

Comment:
I discussed it with @dej611 who has bee the architect of the client side validation logic and we find this change a dangerous one. What we propose is: 

- in esql_editor.tsx just pass {} instead of the callbacks if syntaxValidationOnly is true
- when calling validate you can add a new options argument to control that and forward the syntaxValidationOnly flag.

Meanwhile as there are concerns from our side of disabling this I have raised it with our PM to discuss this on the business level.